fastf1/core.py

```diff
@@ -1884,16 +1885,42 @@ def _check_lap_accuracy(self):
                                           atol=0.003, rtol=0, equal_nan=False)
                     if not check_2:
                         integrity_errors += 1
+                        lap_integrity_flag = False
                 else:
                     check_2 = False  # data not available means fail
 
                 if prev_lap is not None:
+                    if (not pd.isnull(prev_lap['Time'])):
```

Comment:
I don't see any advantage in doing this check here and then setting an additional variable. Keep check_4 self-contained and don't clutter it through the other checks. See other comment below